fiftyone/core/dataset.py

```diff
@@ -1626,6 +1644,526 @@ def add_frame_field(
+
+        index_fields = []
+        summary_info = {"path": path, "index_type": index_type}
+        if index_type == "categorical":
+            summary_info["include_counts"] = include_counts
+            if include_counts:
+                label_field = field_name + ".label"
+                count_field = field_name + ".count"
+                index_fields.extend([label_field, count_field])
+
+                self.add_sample_field(
+                    field_name,
+                    fof.ListField,
+                    subfield=fof.EmbeddedDocumentField,
+                    embedded_doc_type=foo.DynamicEmbeddedDocument,
+                    info={_SUMMARY_FIELD_KEY: summary_info},
+                )
+                self.add_sample_field(label_field, type(_field))
+                self.add_sample_field(count_field, fof.IntField)
+            else:
+                index_fields.append(field_name)
+                self.add_sample_field(
+                    field_name,
+                    fof.ListField,
+                    subfield=type(_field),
+                    info={_SUMMARY_FIELD_KEY: summary_info},
+                )
+        elif index_type == "numeric":
+            summary_info["group_by"] = group_by
+            if group_by is not None:
+                if "." in group_by:
+                    value = group_by.rsplit(".", 1)[1]
+                    group_path = group_by
+                else:
+                    value = group_by
+                    group_path = path.rsplit(".", 1)[0] + "." + group_by
+
+                _group_field = self.get_field(group_path)
+
+                value_field = field_name + f".{value}"
+                min_field = field_name + ".min"
+                max_field = field_name + ".max"
+                index_fields.extend([value_field, min_field, max_field])
+
+                self.add_sample_field(
+                    field_name,
+                    fof.ListField,
+                    subfield=fof.EmbeddedDocumentField,
+                    embedded_doc_type=foo.DynamicEmbeddedDocument,
+                    info={_SUMMARY_FIELD_KEY: summary_info},
+                )
+                self.add_sample_field(value_field, type(_group_field))
+                self.add_sample_field(min_field, type(_field))
+                self.add_sample_field(max_field, type(_field))
+            else:
+                min_field = field_name + ".min"
+                max_field = field_name + ".max"
+                index_fields.extend([min_field, max_field])
+
+                self.add_sample_field(
+                    field_name,
+                    fof.EmbeddedDocumentField,
+                    embedded_doc_type=foo.DynamicEmbeddedDocument,
+                    info={_SUMMARY_FIELD_KEY: summary_info},
+                )
+                self.add_sample_field(min_field, type(_field))
+                self.add_sample_field(max_field, type(_field))
+
+        if sidebar_group is not False:
```

Comment:
Is there an advantage in not having a sidebar group for field summary? We can simplify by just having a `summaries` group by default

Reply:
The default *is* to create a `summaries` group by default, Passing `sidebar_group=False` is just a bail out option in case the user doesn't want us to touch the sidebar groups at all.

For context, `dataset.app_config.sidebar_groups=None` by default and is only lazily initialized if the user actually drags fields around in the App (or modifies it via the SDK). This is exceptionally niche, but providing `sidebar_group=False` as an option allows for a case where a user prefer for `dataset.app_config.sidebar_groups` to stay as `None`

Reply:
I see, makes sense. Just a hypothetical thought, if we don't allow `sidebar_group=False`, the user has to clear the side bar groups manually through SDK, or UI?

Reply:
If sidebar groups exist and the user doesn't like them, then yes they either have to drag-to-reorder in the App or they could reset via the SDK if desired like so:

```py
dataset.app_config.sidebar_groups = None
dataset.save()
```